Added network policy trouble shooting guide (#146)
Maistra/maistra.github.io: topics/docs/troubleshooting/troubleshooting_network_policy.adoc

---
title: "Network Policy Issues"
type: "document"
category: "Troubleshooting"
description: "This document describes common issues experienced with regards to Network Policies."
---

=== Routes Not Working

> If you add a project to the data plane, pods of this project might be no longer accessible from the OpenShift Ingress or Router, depending on your network polices. Since adding a project to a data plane, a network rule will also be added that allows the OpenShift Ingress or Router to access only pods with the label `maistra.io/expose-route: 'true'`.

Reviewer: ```suggestion
If you add a project to the mesh, pods of this project might be no longer accessible from the OpenShift Ingress or Router, depending on your network polices. When adding a project to a mesh, a network rule will also be added that allows the OpenShift Ingress or Router to access only pods with the label `maistra.io/expose-route: 'true'`.
```

> For deployments you add the label `maistra.io/expose-route: 'true' `to the pods with the following command:

Reviewer 2: Thanks for your contribution. 

Small comment, this looks like it didn't render properly:

![image](https://user-images.githubusercontent.com/8484260/83528864-9db3d400-a4b7-11ea-8df1-1be9e67ac4ce.png)

Author: Hello Brian,

Sorry I missed that point. I have to admit that I just used the preview in my code editor, because I didn't want to install hugo and the other components. Also, I noticed that the link to the troubleshooting page was incorrect. I fixed both errors. 

After your feedback I created for me a container which allows me to verify my changes locally. Should I create a pull request for this, so that other users can use this container as well?

Kind regards,

Olaf

Reviewer 2: We have a build container that we use which has all of the tooling in it already. I can look into combining that with our makefile in order to allow the commands to all be run through docker without the installation of additional tooling. I've created [Maistra-1543](https://issues.redhat.com/browse/MAISTRA-1543) to track it.

Reviewer 2: This is our image: https://quay.io/repository/maistra-dev/maistra-builder?tab=tags. We have tags for Maistra 1.1 and Maistra 1.2.

> For deployments you add the label `maistra.io/expose-route: 'true'` to the pods with the following command:

Reviewer: ```suggestion
If you're using Deployments, you can add the label `maistra.io/expose-route: 'true'` to the pods with the following command:
```

[source,bash]
----
oc patch deployment <deployment name> -p '{"spec":{"template":{"metadata":{"labels":{"maistra.io/expose-route":"true"}}}}}'
----

> In certain cases i.e run the OpenShift cluster on VMWare, you might need to add the label `network.openshift.io/policy-group: ingress` to the default namespace. A detailed description can be found here: https://docs.openshift.com/container-platform/4.4/networking/configuring-networkpolicy.html

Reviewer: ```suggestion
In certain cases (e.g. when running the OpenShift cluster on VMWare), your ingress controller is running on the host network. In this case, ingress traffic will appear to originate in the `default` namespace. In this case you might want to add the label `network.openshift.io/policy-group: ingress` to the default namespace. A detailed description can be found here: https://docs.openshift.com/container-platform/4.4/networking/configuring-networkpolicy.html
```

> In certain cases i.e run the OpenShift cluster on VMWare, you might need to add the label `network.openshift.io/policy-group: ingress` to the default namespace. A detailed description can be found here: https://docs.openshift.com/container-platform/4.4/networking/configuring-networkpolicy.html

Reviewer 2: ```suggestion
In certain cases i.e run the OpenShift cluster on VMWare, you might need to add the label `network.openshift.io/policy-group: ingress` to the default project. A detailed description can be found here: https://docs.openshift.com/container-platform/4.4/networking/configuring-networkpolicy.html
```
